README.md

```diff
@@ -422,6 +432,53 @@ When storing logs on disk, we recommend using Logrotate in truncate mode.
 
+## Suffixing `tree_path`
+
+Kiev is built upon the assumption that one request is only handled once. This isn't always true.
+
+A practical example: one Amazon SNS topic with a few Amazon SQS queues subscribed to it. You send one message to SNS and queues receive identical copies that are impossible to distinguish in the trace without any help from the outside.
+
+You can solve this by adding a fixed unique suffix inside each queue processor. Preferably a single character with an even number in the alphabet (B, D, F and so on), to maintain the notion of "asynchronous processing" used throughout Kiev.
+
+For a combination of SNS and [Shoryuken](https://github.com/phstc/shoryuken) (SQS consumer). Here's how you can use it:
+
+* Ensure that "Raw Message Delivery" is enabled in your SQS-to-SNS subscriptions
+* On sender, write `Kiev::SubrequestHelper.payload` into the message attributes
+* On each receiver, use `Kiev::Shoryuken::suffix_tree_path` with a unique tag, like this:
+
+  ```ruby
+  # Suffix a single worker class:
+  class MyWorker
+    include Shoryuken::Worker
+    Kiev::Shoryuken.suffix_tree_path(self, "B")
+    # ...
+  end
+
+  # Or use a suffix process-wide:
+  Shoryuken.configure_server do |config|
+    Kiev::Shoryuken.suffix_tree_path(config, "B")
+  end
+  ```
+
+Here's an example of the possble `tree_path` sequence you could get by configuring two consumers with suffixes `1` and `2` (note ordering by `tree_path`):
+
+| `tree_path` |                             Meaning                             |
+|-------------|-----------------------------------------------------------------|
+| `A`         | An entry point into the system, a synchronous request           |
+| `AB`        | Background job caused by `A` executed                           |
+| `ABA`       | Synchoronous request made from `AB`                             |
+| `ABD`       | _(Not logged by Kiev itself)_ `AB` sends out an SNS message     |
+| `ABD1`      | Message `ABD` handled by susbcriber `1`                         |
+| `ABD1A`     | Synchronous request sent by `1` when handling the message `ABD` |
+| `ABD1C`     | Synchronous request sent by `1` when handling the message `ABD` |
+| `ABD2`      | Message `ABD` handled by susbcriber `2`                         |
+| `ABD2A`     | Synchronous request sent by `2` when handling the message `ABD` |
+| `ABF`       | Another backgound job from `AB` executed                        |
+| `AD`        | Background job caused by `A` executed                           |
+| `AE`        | Synchronous request made from `A`                               |
+
+Without numbers you have no way of immediately knowing who made the request `ABDC` and you'd have two entries for both `ABD` and `ABDA`. As different subscribers may log different fields, you might be able to tell apart `ABD` events. But `ABDA` events could happeen in the same place and have the same fields and would be practically the same.
```

Comment:
> Without numbers you have no way of immediately knowing who made

Without numbers you have no way to know who made

Reply:
Reworded into:

> Without numbers you won't see at a glance who made

...it might be possible, but would require a closer look at the records.